openpaas-james/mailbox/encrypted/memory/src/main/java/com/linagora/openpaas/encrypted/InMemoryEncryptedMailboxManager.scala

```diff
@@ -0,0 +1,123 @@
+package com.linagora.openpaas.encrypted
+
+import java.util.Optional
+import java.{lang, util}
+
+import javax.inject.Inject
+import org.apache.james.core.Username
+import org.apache.james.mailbox.MailboxManager.{MailboxCapabilities, MailboxRenamedResult, MessageCapabilities, SearchCapabilities}
+import org.apache.james.mailbox.model.search.MailboxQuery
+import org.apache.james.mailbox.model.{Mailbox, MailboxACL, MailboxAnnotation, MailboxAnnotationKey, MailboxId, MailboxMetaData, MailboxPath, MessageId, MessageRange, MultimailboxesSearchQuery}
+import org.apache.james.mailbox.{MailboxManager, MailboxSession, MessageManager}
+import org.reactivestreams.Publisher
+import reactor.core.publisher.Flux
+
+class InMemoryEncryptedMailboxManager @Inject()(mailboxManager: MailboxManager) extends MailboxManager {
```

Comment:
EncryptedMailboxManager

Reply:
`
override def getMailbox(mailboxPath: MailboxPath, session: MailboxSession): MessageManager = new EncryptedMessageManager(mailboxManager.getMailbox(mailboxPath, session), new InMemoryKeystoreManager())
`
What if we want to use Cassandra version of the keystore?

Reply:
Then we inject the CassandraMailboxManager (decorator pattern)

Reply:
InMemoryEncryptedMailboxManager -> EncryptedMailboxManager